core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/provider_life_cycle.go

```diff
@@ -147,7 +146,9 @@ func (p *logEventProvider) register(ctx context.Context, lpFilter logpoller.Filt
 func (p *logEventProvider) UnregisterFilter(upkeepID *big.Int) error {
 	// Filter might have been unregistered already, only try to unregister if it exists
 	if p.poller.HasFilter(p.filterName(upkeepID)) {
-		if err := p.poller.UnregisterFilter(p.filterName(upkeepID)); err != nil {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		if err := p.poller.UnregisterFilter(ctx, p.filterName(upkeepID)); err != nil {
```

Comment:
Thankfully, caller has a context so you can just pass that `r.ctx`

Reply:
Sweet, I'll pass that down